src/libraries/System.Net.Quic/src/System/Net/Quic/NetEventSource.Quic.Counters.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Diagnostics.Metrics;
+using System.Net.Quic;
+
+using Microsoft.Quic;
+using static Microsoft.Quic.MsQuic;
+
+namespace System.Net
+{
+    internal sealed partial class NetEventSource
+    {
+        private static Meter s_meter = new Meter("Private.InternalDiagnostics.System.Net.Quic");
+        private static readonly long[] s_counters = new long[(int)QUIC_PERFORMANCE_COUNTERS.MAX];
+
+        public static readonly ObservableGauge<long> MsQuicCountersGauge = s_meter.CreateObservableGauge<long>(
+            name: "MsQuic",
+            observeValues: GetGauges,
+            unit: null,
+            description: "MsQuic performance counters");
+
+        public static readonly ObservableCounter<long> MsQuicCountersCounter = s_meter.CreateObservableCounter<long>(
+            name: "MsQuic",
+            observeValues: GetCounters,
+            unit: null,
+            description: "MsQuic performance counters");
+
+        [NonEvent]
```

Comment:
Any practical reason the counter logic is baked into the `NetEventSource` class instead of living in a separate one?

Reply:
I first put it there because I started with the older API and with Metrics API there is no need for the code to be in `NetEventSource` specifically. But I would probably keep it avoid having to initialize the class explicitly (the cctor is guaranteed to run because `NetEventSource` is used from MsQuicApi class).

Reply:
@ManickaP what do you think, should I move the code to a separate class to avoid polluting the NetEventSource with global statics for the instruments?

Reply:
Looking at Http space, metrics seem to be completely separate. But I don't have a strong opinion here and don't mind it either way.

Reply:
ok, I will keep it here to avoid the need for explicit initialization hook.